sdks/agent-sdk/src/core/Agent.ts

```diff
@@ -0,0 +1,128 @@
+import EventEmitter from "node:events";
+import type { ContentCodec } from "@xmtp/content-type-primitives";
+import {
+  Client,
+  type ClientOptions,
+  type DecodedMessage,
+} from "@xmtp/node-sdk";
+import { filter } from "@/utils/filter";
+import { AgentContext } from "./AgentContext";
+
+interface EventHandlerMap<ContentTypes> {
+  error: [error: Error];
+  message: [ctx: AgentContext<ContentTypes>];
+  start: [];
+  stop: [];
+}
+
+export interface AgentOptions<ContentTypes> {
+  client: Client<ContentTypes>;
+}
+
+export type AgentMiddleware<ContentTypes> = (
+  ctx: AgentContext<ContentTypes>,
+  next: () => Promise<void>,
+) => Promise<void>;
+
+export class Agent<ContentTypes> extends EventEmitter<
+  EventHandlerMap<ContentTypes>
+> {
+  readonly client: Client<ContentTypes>;
```

Comment:
this can be `#client`, then you can add a getter for the value.

```
get client() {
  return this.#client;
}
```

Reply:
Ok, I did that also for the `AgentContext`.